frontend/src/app/offers/create-offer/create-offer.component.ts

```diff
@@ -0,0 +1,80 @@
+import { Component, OnInit } from '@angular/core';
+import { Offer } from 'app/homepage-offer/offers.model';
+import { AuthService } from 'app/auth.service';
+import { User } from 'app/user';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+import { OffersService } from 'app/homepage-offer/offers.service';
+import { ActivatedRoute, Params } from '@angular/router';
+import { environment } from 'environments/environment';
+
+@Component({
+  selector: 'volontulo-create-offer',
+  templateUrl: './create-offer.component.html',
+  styleUrls: ['./create-offer.component.scss']
+})
+export class CreateOfferComponent implements OnInit {
+  djangoRoot: string;
+  public inEditMode: boolean = false;
+  offer: Offer = new Offer;
+  user: User;
+  isAdmin: boolean = false;
+  hasOrganization = false;
+
+  constructor(
+    private authService: AuthService,
+    private offersService: OffersService,
+    private route: ActivatedRoute,
+  ) {
+      this.djangoRoot = environment.djangoRoot;
+  }
+
+  ngOnInit() {
+    this.authService.user$
```

Comment:
Please remember to unsubscribe this subscription. You can do it in `ngOnDestroy`. This applies to all the subscriptions that you manually create and they do not complete (i.e. HTTP requests do complete). You don't have to do it for router subscriptions (like the one in line 44).

Reply:
BTW, I'm not sure if these subscriptions are not unavoidable (in favour of just defining observables and subscribing with `async` pipes).